### PR TITLE
fix: installed apps and admin apps layout spacing issues.

### LINKED ISSUE #6735: [CAL-939] [CAL-789] Installed Apps: Layout, spacing issues
Currently

Should be

View in Figma

[📐SPEC: Installed Apps - Cal - Live (Figma)](https://www.figma.com/file/xk4HOxtSI82J0F7enMxeak/Cal---Live?node-id=1256%3A108316&t=2RhzAobvSiPXOBD0-1)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-789](https://linear.app/calcom/issue/CAL-789/installed-apps-layout-spacing-issues)</sub>

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-789](https://linear.app/calcom/issue/CAL-789/installed-apps-layout-spacing-issues)</sub>

### PR DESCRIPTION
**Pull request by G3root**
## What does this PR do?

Fixes #6735 

things done in this PR:

- [x] refactor `AppListCard` component to make it highlightable and add some props from `IntegrationListItem` component.
- [x] replace  `IntegrationListItem` component with `AppListCard` component.
- [x] fix layout issues in admin app list page
- [x] fixed layout issues in `apps/installed`  
- [x] refactor: remove  `IntegrationListItem` component

## before admin apps:
Screenshot 2023-01-27 at 13-59-13 Apps Cal com

## after admin apps:
Screenshot 2023-01-27 at 13-57-38 Apps Cal com

## before installed apps
Screenshot 2023-01-27 at 15-03-18 Installed Apps Cal com

## after installed apps

Screenshot 2023-01-27 at 15-02-08 Installed Apps Cal com

**Environment**: Staging(main branch) / Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: apps/web/components/auth/layouts/AdminLayout.tsx [hunk 1]
```diff
@@ -22,10 +23,11 @@ export default function AdminLayout({
     }
   }, [session, router]);
 
+  const isAppsPage = router.asPath.startsWith("/settings/admin/apps");
```
Comment (G3root) [inline apps/web/components/auth/layouts/AdminLayout.tsx:26]: workaround to apply styles only in `/settings/admin/apps` page

File: apps/web/pages/apps/installed/[category].tsx [hunk 2]
```diff
@@ -80,11 +80,7 @@ function ConnectOrDisconnectIntegrationButton(props: {
   }
   /** We don't need to "Connect", just show that it's installed */
   if (isGlobal) {
-    return (
-      <div className="truncate px-3 py-2">
-        <h3 className="text-sm font-medium text-gray-700">{t("default")}</h3>
-      </div>
-    );
+    return null;
```
Comment (G3root) [inline apps/web/pages/apps/installed/[category].tsx:83]: show nothing if it is a global app

File: apps/web/components/AppListCard.tsx [hunk 3]
```diff
@@ -1,35 +1,93 @@
...
+  const router = useRouter();
+  const [highlight, setHighlight] = useState(shouldHighlight === true && hl === slug);
+  const timeoutRef = useRef<NodeJS.Timeout | null>(null);
+
+  useEffect(() => {
+    if (shouldHighlight === true && highlight) {
+      const timer = setTimeout(() => {
+        setHighlight(false);
+        const url = new URL(window.location.href);
+        url.searchParams.delete("hl");
```
Comment (G3root) [inline apps/web/components/AppListCard.tsx:51]: remove `hl` query parameter after highlighting

**Review by PeerRich — APPROVED** [review]
great work! tested and works

**Comment by emrysal** [comment]
Should this toggle be at the end? @Jaibles 

image

**Comment by Jaibles** [comment]
> Should this toggle be at the end? @Jaibles
> 
> image

No - I have no toggle in the designs. Assuming it's to uninstall? If so, we need to change that.
CleanShot 2023-01-27 at 13 24 16@2x

**Comment by Jaibles** [comment]
Also, there are a lot of UI and layout issues on the page. The main thing is the space on the right that should not be there, but also within the app cards etc.

CleanShot 2023-01-27 at 13 27 48@2x

[Please refer to the designs. They are spec'd in detail](https://www.figma.com/file/xk4HOxtSI82J0F7enMxeak/Cal---Live?node-id=1256%3A108316&t=tFVJMR9veFYT6fmC-1)

**Review by emrysal — CHANGES_REQUESTED** [review]
@Jaibles found more differences with the design, could you include this with this (so far great!) PR @G3root ?

**Comment by G3root** [comment]
new changes.

- [x] use modal for editing keys.
- [x] use dropdown for action buttons.
- [x] add disabled style for `DropdownItem` component.
- [x] some refactors.

 
[Screencast 2023-01-30 18:57:53.webm](https://user-images.githubusercontent.com/84864519/215491444-157b0d85-6fae-4914-b39b-b5137dbd9a98.webm)

**Review by leog — DISMISSED** [review]
Looks good! @G3root would you mind test the admin wizard onboarding version of the admin apps list? I know I made changes to adapt the UI a bit in my PR #6574, just wanted to see if your changes go well with mine. You can see that signing in as an admin and visiting /auth/setup. It's the last step.

**Comment by G3root** [comment]
@leog here's how the onboarding look's like in this branch.

[Screencast 2023-01-30 22:10:37.webm](https://user-images.githubusercontent.com/84864519/215540029-e1215bde-ff7a-4064-acff-1bd6fa2a0d0a.webm)

File: apps/web/components/AppListCard.tsx [hunk 4]
```diff
@@ -1,35 +1,93 @@
...
+    shouldHighlight,
+    isTemplate,
+    invalidCredential,
+    children,
+  } = props;
+  const {
+    data: { hl },
+  } = useTypedQuery(schema);
+  const router = useRouter();
+  const [highlight, setHighlight] = useState(shouldHighlight === true && hl === slug);
```
Comment (alannnc) [inline apps/web/components/AppListCard.tsx:43]:
```suggestion
  const [highlight, setHighlight] = useState(shouldHighlight && hl === slug);
```

File: apps/web/components/AppListCard.tsx [hunk 5]
```diff
@@ -1,35 +1,93 @@
...
+  const timeoutRef = useRef<NodeJS.Timeout | null>(null);
+
+  useEffect(() => {
+    if (shouldHighlight === true && highlight) {
```
Comment (alannnc) [inline apps/web/components/AppListCard.tsx:47]:
```suggestion
    if (shouldHighlight && highlight) {
```

File: apps/web/components/AppListCard.tsx (hunk 3 above)
Comment (alannnc) [reply apps/web/components/AppListCard.tsx:51]: Is this required? Probably losing hl if reloading.

**Review by emrysal — APPROVED** [review]
Looks Good To Me 🚀

**Pull request merged**